src/test/java/com/brightcove/castlabs/client/CastlabsClientTest.java

```diff
@@ -270,8 +257,76 @@ public void testIngestKeysErrorNotFound() {
             fail(e.getMessage());
         }
 
-        mockServerClient.verify(loginRequest, VerificationTimes.once());
-        mockServerClient.verify(ticketRequest, VerificationTimes.once());
         mockServerClient.verify(ingestAssetKeysRequest, VerificationTimes.once());
     }
+
+    @Test
+    public void testItCanMakeASubMerchantCreationRequest() throws Exception {
+        final String merchantId = "merchX";
+        final HttpRequest expectedRequest =
+                request().withMethod("POST").withPath("/frontend/rest/reselling/v1/reseller/" + merchantId + "/submerchant/add")
+                        .withQueryStringParameter("ticket", exampleTicket)
+                        .withHeader("accept", "application/json")
+                        .withHeader("content-type", "application/json");
+        final String mockResponse = IOUtils.toString(new FileInputStream("src/test/resources/sample_sub_merchant_creation_response.json"));
+        mockServerClient.when(expectedRequest).respond(response().withStatusCode(200).withBody(mockResponse));
+        AddSubMerchantAccountRequest request = new AddSubMerchantAccountRequest();
+        request.setApiNameSuffix("exampleAPINameSuffix");
+        request.setLinkedUserUuid("10f38484-ff73-4101-b879-8794dd26fe77");
+        request.setName("Example Name");
+
+        final AddSubMerchantAccountResponse response = castlabsClient.addSubMerchantAccount(request, merchantId);
+
+        assertNotNull(response);
+        assertEquals(response.getSubMerchantUuid(), "0491caa4-8392-4b42-badc-f38a00134d99");
+        mockServerClient.verify(expectedRequest, VerificationTimes.once());
+    }
+
+    @Test
+    public void testItCanHandleCastlabsErrorsWhenMakingASubMerchantCreationRequest() throws Exception {
+        final String merchantId = "merchX";
+        final HttpRequest expectedRequest =
+                request().withMethod("POST").withPath("/frontend/rest/reselling/v1/reseller/" + merchantId + "/submerchant/add")
+                        .withQueryStringParameter("ticket", exampleTicket)
+                        .withHeader("accept", "application/json")
+                        .withHeader("content-type", "application/json");
+        final String mockResponse = IOUtils.toString(new FileInputStream("src/test/resources/sample_sub_merchant_creation_error_response.json"));
```

Comment:
We can remove the need to care about where the resource directory is by using `this.getClass().getClassLoader().getResource("foo.json").getFile()`.